UCT/DC/GTEST: RDMA_READ flow control

brminich opened the pull request:
## What
RDMA READ operations flow control for dc transport
rc part was implemented in #4716

## Why ?
Can be useful to avoid/control congestions

## How ?
Maintain a counter of simultaneous get operations

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 07-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/3551e664d5ac9aa8a6d59bd07ffaed9c079d33b5/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/5807c6d7a13eb7f00345e2da455efba42ccc6636/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/c375c2e7a5367c40251291c4fd31c370da6062bd/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/34ed9b240a351e33c5fd504e3be770cb11dccf12/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/485ddc00effa4b93e9414dc2f7d6fb948b2a367d/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/68d5cff82a28973e64db3016486c4aca6d7043bb/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/a40b856af4ca8ecfcc341d38c19368f859b1780a/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/1eff3c6cf548b89b616198859eb24ea17ec5c6cd/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/3185927eef41df7cbe49419c47f1d622c380ca05/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/687cab57fb63babe5740de519ea5df59306f4caf/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/38c277ffef5e34570e68983c0f7ac2c655996794/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/9a5e6b202b5b52d5b1a9bb254f2562e8d6d49213/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/87e3c231a39aa3aabfdec409b71c40d7e06e53b9/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/6b94e7d86e61ce1fc553b550de4e56cdff820f68/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/8d512b8a5824a45c2ca5ea1ba7feafb5b12a2c15/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/9c877eb7ea481315cec5df68bf0035682d93ac0a/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/3176eda7d36e9c317bc22576f212e83b9936106b/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/17098b20004a072d6c2be47f99b6d41d7a6429d7/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/74b63152c9c4a40acc8fde6d4eb5076701cb95ee/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/77f8293eac7f971d5a5fcdb56cd823065f6d9195/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/78b63ed7f9cebb6107dde0039f8830d4c4a86b38/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/e2170f32517df8ba2699d7f5bfded795b5723d94/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/b63f6bfbee4b88793e44aa1006ac0f407b7e0d10/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/6154ad082cf13b5e0d15ba8f137ba0f916fa0f86/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/85947ec284084b6ed75d51cb7bc5a09d/raw/e77fb7f536dcae6c22fac787c8e68040c1ac39da/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

yosefe commented on src/uct/ib/dc/dc_mlx5.h:
```diff
@@ -293,7 +293,8 @@ static inline uint8_t uct_dc_mlx5_iface_dci_find(uct_dc_mlx5_iface_t *iface, uin
 {
-    return !ucs_mpool_is_empty(&iface->super.super.tx.mp);
+    return !ucs_mpool_is_empty(&iface->super.super.tx.mp) &&
+           (iface->super.super.tx.reads_available != 0);
```
maybe > 0 ?

yosefe commented on src/uct/ib/dc/dc_mlx5_ep.h:
```diff
@@ -435,6 +435,10 @@ static inline ucs_status_t uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface,
     ucs_assert(!iface->super.super.config.tx_moderation);
 
+    if (!uct_dc_mlx5_iface_has_tx_resources(iface)) {
+        goto out_no_res;
```
seems same as line 448?
should we count stats here?

brminich replied:
no, below we check for DCI resources by `uct_dc_mlx5_iface_dci_has_tx_resources`
will add stats

yosefe commented on src/uct/ib/dc/dc_mlx5_ep.h:
```diff
@@ -536,7 +540,9 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
-                    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(_iface, (_ep)->dci), \
+                    ucs_assertv_always(uct_dc_mlx5_iface_dci_has_outstanding(_iface, \
+                                                                             (_ep)->dci) || \
+                                       !uct_dc_mlx5_iface_has_tx_resources(_iface) , \
```
why it is needed ? maybe epplain in the comment in line 531-534?

brminich replied:
fixed this problem by another way

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 7 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/60b5f23e768cadda77b61f9c95c8ad6f986a6596/_main) | :x: FAILURE
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/46e42e6ac08709c41c8dc972eeabc57c6bfd2870/hpc-arm-cavium-jenkins_W1) | :x: FAILURE
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/90721649acc0d3126e411f1297ef4fbfe50542e2/hpc-arm-hwi-jenkins_W1) | :x: FAILURE
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/9c979203f7c722a77c9b95350b43c3231a7870f9/hpc-test-node-gpu_W1) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/863b12e04009b289a913b20eabe8b9c6b4aa8cb0/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/f4fcdbac8fb596aa6a1bcb301b3c6520a6d95573/hpc-test-node-new_W1) | :x: FAILURE
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/88d159d07d8b4fb7d4bfb6fd81ab2417ef04fb18/r-vmb-ppc-jenkins_W1) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/0bcf5d2ff5c21c130f26c5a7ab1b4b3bfe1bde88/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/31553a772082dddb92664d67ba57386f4c7502a3/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/8d761e472cc9066160f333a9a397dee66553a4bb/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/4e0e0ced2300bd35ba6830efa23358e44275afe6/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/79a409a2c361c1ce7fc82dcf5f496c78f97b66b5/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/ddea76563ce783d15fa2f7670923cdc10bf87016/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/c644c80f4cc0188daac5617e00652144120a0720/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/b57a13ab2ea05ba122d9e140b52098a700b53bb4/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/e830603e461e79d00c8d66936cab68e9b79cac9f/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/3a645ca7cf40afbed23d1d079da43d82d798189d/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/c8432889414758796b7961ac345b27cc43f6269b/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/4dc9b0c182fd9c5b1ed1250b878512bdecdea683/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/835978a8a3d50d6e3e948ec27905abc298a02a96/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/a0a126cb14442297ca837915c0d4995d805ced1e/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/a24eeb25a64462ddc95ce02e86e0012ca1521c0f/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/698cfd258c84630bc29a33071a766020ad74fd11/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/c77505801b4b3a4815a138ef76f08e6c28c0269b/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/97ba330cd8c8073d59b508eb004b5a3d/raw/6a8a4fa3891902b7f7cc4985bed67255f5005fe4/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/c10d01d4092afae7fd2a2dd8686866ede7d215d7/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/0fc1e54823e84caa293e0bf9ca915453794e405d/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/232b3d3d96e3bffb4dd25c306a59d7137b5305e1/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/df50e249b1e429d293e9a6e24de65a29a016719a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/a5c78e6d61311b92b22bf62c4252b9032637340d/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/ad8f4c377e4feefe1096825c91c6c6816057a551/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/0b25ea482497514298e2cffec7f6444c351da6a6/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/e2c85d378b7eabca48d5f5b29d7e1816094ca05e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/9f96444bdb5514ebcd7a1939b49d0b58a17fbdbe/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/f774f39209b4f9f164fdc424fdb3896c16686767/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/05ec627dc872cd586fd51761d0a4697dbabec034/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/bc8c704dd7406d83156e1260dc2f456619d29bb8/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/fe97dbbcbabc9decb1a94cc054e42d28543eb5e4/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/6759f7f7803ebf2493687e52a97d7520e238f20e/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/71ba2e2c12eaee1aedd8928522a886f44cafd5c0/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/8b37c0006a532b01aae698e4fee5f62827aa4f23/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/22f067a30bf60ca2283c631c62e04ca858f88695/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/fe31de587e542c2ca6bbca00b0e5f16a9a0b4d36/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/bdcb39642434d0be0e12a9c15bcc7a460ef54728/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/ba2578cbe01987372d28aa4701615cf55eddd8bf/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/61df405c624f84ba023ac670adedc7e8c154addb/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/28008c1ef8b8d18ca243502c3625bc690f33c422/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/ca25a5a94ede2c783a487b0219477bf379736665/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/a03b995b9b740232ea63652e1c09fbf950ca0368/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/6edbdff7c7841015e621435ff6987a15/raw/c6425e04b073efa3d2b4878ce1856e072d942a44/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

brminich commented:
@yosefe, fixed

brminich commented:
@hoopoepg, @dmitrygx 
plz review

dmitrygx commented on src/uct/ib/dc/dc_mlx5_ep.h:
```diff
@@ -508,7 +516,15 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
+        } \
+    }
+
+#define UCT_DC_MLX5_CHECK_DCI_RES(_iface, _ep) \
```
what this macro is named as "CHEC_DCI_RES" if we don't check iface res?

brminich replied:
dci is dci initiator (QP), we check just QP resources

yosefe commented on test/gtest/uct/ib/test_rc.cc:
```diff
@@ -144,17 +147,34 @@ class test_rc_get_limit : public test_rc {
         check_skip_test();
     }
 
+#if ENABLE_STATS
```
do we really need this '#if' ?

brminich replied:
in theory we could build tests without stats

yosefe replied:
would it really fail to compile in this case? seems the code here would just do nothing if no stats

brminich replied:
i'll check, but imo better have it in any case as "documentation"

brminich replied:
it fails with
```
common/test.cc:104: Failure
Failed
Invalid UCS configuration for STATS_DEST : file:/dev/null, error message: No such element(-12)
common/test.cc:104: Failure
Failed
Invalid UCS configuration for STATS_TRIGGER : exit, error message: No such element(-12)
uct/uct_test.cc:581: Failure
Value of: ucs_stats_is_active()
  Actual: false
Expected: true
[jazz05:125628:0:125628]       test.cc:246  Assertion `m_state == FINISHED || m_state == SKIPPED || m_state == ABORTED' failed: state=1

```

yosefe commented on test/gtest/uct/ib/test_rc.cc:
```diff
@@ -144,17 +147,34 @@ class test_rc_get_limit : public test_rc {
     }
 
     void init() {
+#if ENABLE_STATS
```
do we really need this '#if' ?

yosefe commented on src/uct/ib/dc/dc_mlx5_ep.h:
```diff
@@ -508,7 +516,15 @@ static inline struct mlx5_grh_av *uct_dc_mlx5_ep_get_grh(uct_dc_mlx5_ep_t *ep)
 #define UCT_DC_MLX5_CHECK_RES(_iface, _ep) \
     { \
-        ucs_status_t status = uct_dc_mlx5_iface_dci_get(_iface, _ep); \
+        ucs_status_t status = uct_dc_mlx5_iface_dci_get(_iface, _ep, 1); \
```
so it means that now all AM functions except flow control will have extra overhead for checking iface resources?

brminich replied:
yes we will check for iface resources (rdma reads)

yosefe replied:
why do we have to add a check for rdma reads and mpool_is_empty in am_short flow?

brminich replied:
to preserve ordering

yosefe replied:
but it adds extra overhead
and we don't really need to preserve ordering between rdma_read and am's. for example we don't check FC for rdma_read.

yosefe commented:
also, there are several code style fixes in this PR. Pls separate them to another PR next time

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 13-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/d0ef8e5294616ac689d4f24513382e5c7f026210/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/8f804322bfffc36f41afedaa782b21fc087d806b/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/fa672c4141578670bc458c1c1c42f387f6273340/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/55612a607443e235bdf014978940c596c9bf26ac/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/bc0fb1e38123537933e1f01923a7e705ee658192/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/0c81a86fcc4c1ed63d832c5be85daa5835810f1f/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/e4c896aca424f33903330df1aa5e07987e1d9914/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/dbae7b869a578af62ad213839924af2b0a7b3761/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/b2d8d39e3958fb62fce87e234e7e2cae13af40bf/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/cef7fad2955d58e9c75a210817741f9e5eede908/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/33a50077a3512731ff0c7309cf921f22eb4361e4/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/cd76bd1a0c470535c3d380d2908b158b9a9475db/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/b8dc8f3667aeb018b0827a577dcfb0309df9b64e/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/1224ef1e363b75d57cf0b601a510cc1d95de193e/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/d6a4de099da3ce507dca7ef5c39f92c66e14f5e8/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/d42b6ae3d03c714dd479774845203e6d26057136/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/dc78a2d90fafd1de845b531d4b93b9218b696a09/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/a6384613de641d960d519d8a56d1208bf5992d6a/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/69bf2666f79cc2059ca1fdd06b25281694a5891b/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/80862cde5c7084dd56392491759b2c0f5427faad/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/883775b5533e1e352591456ddc7f60270dcb7f2b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/742034590d26f28478d323f863ed42ab23970e6a/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/6159773d6eea95d41a4d20682fdf7fa217e4ee62/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/fb9e788262e557c3b5c1ce082ec041663b742b0b/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/056827489680c7e579d7182a488d09ef/raw/cd531b2422807a07b835bc09efc7a79927caa7fd/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

yosefe commented:
@brminich seems like put_short/bcopy/zcopy still check rdma_reads 
i think that instead of changing existing functions/macros for DC, just add read resources check for get_bcopy and get_zcopy, like you did for RC with UCT_RC_CHECK_NUM_RDMA_READ()

brminich commented:
> @brminich seems like put_short/bcopy/zcopy still check rdma_reads
> i think that instead of changing existing functions/macros for DC, just add read resources check for get_bcopy and get_zcopy, like you did for RC with UCT_RC_CHECK_NUM_RDMA_READ()

this way we will break no-op fence

yosefe commented:
> this way we will break no-op fence

So it means RC is broken for now?

brminich commented:
yes

brminich commented:
will address comments, when #4865 is merged (to reuse a macro and run newly added test)

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 20-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/debc81107459a90a9b56bd0b6a267d00aef014f8/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/bf8fc7da590647c05ef229be5341f0b97b109f4a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/f3eaaa3888738c2bef33942d49b915f61575923f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/a268b3f4acb4a52bfd3ac5b0723e1269e00af366/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/5c90152472ca99fa2190c42ad411e368fadfc592/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/67e43224eee187a4cd1c11a90d9c7797e575e584/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/9465b69e38ba63b634aa423a065dfe918c21c913/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/2796b74841464e81cef2de48115557d0f08d674e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/ced18f86199528013a839a364b7d9144e946d7a3/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/ec07f5798e2fc812d23f410d36145802eb3b14c5/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/f3e319324db8027e89d705af33f49573d265df8d/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/74e76289321834a42b7055e72381d89e01ed6a83/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/d2c55b5d3863a12239785908f7871a645684cdf4/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/afbb68cb08a2788f48b565051af863ad004ae43a/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/a3e9c6897a943f4bad244ceab7fdd84bd6a8ebb2/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/1b00c7ccc1ef1ef5382e8d286ff2a0573817012d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/33c2119ddbebb93c67943ced608d66f70297c0e9/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/dc5f668058cdd82ef75059ea0452bc32c404d3a5/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/32baf877e91084e27e60396e306cc13183f6a3e5/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/0df09f384c1389eedb86e5c4c0ac34447526045e/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/0bb29cff46fd5b15bc17ff19f9c21af2ca1ba3ee/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/6e6a2add1c01a8472968ee126b7068b019de9ad3/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/e1c0b0e124ebff72fdfd30c0ca63f3c4ee5fec9e/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/d93a061ba7c55d31b737ff0f2dcced9fddce5bd5/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/655971ec267962737b39b5d8c6830b9d/raw/205960fd568a3d8aa4aae7e82d3d4aead505e70a/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

brminich commented:
> @brminich why are you changing test/gtest/uct/test_tag.cc in this PR?
> is this leftover from other PR?

I introduced new UCT test create_entity method, which creates tag offload capable interface. It is needed, because recently we made a change that tag offload is not initialized if no eager and rendezvous callbacks passed to the iface create function.
 So, I just removed code duplication.

yosefe commented:
> I introduced new UCT test create_entity method, which creates tag offload capable interface. It is needed, because recently we made a change that tag offload is not initialized if no eager and rendezvous callbacks passed to the iface create function.
> So, I just removed code duplication.

So, why need to do tag operations in the test case which checks for RMA thresholds?
if we want to check other operations are not affected, wouldn't it be better to check AM?

brminich commented:
Initially I did not change tag operation checks in DC and they were blocked by lack of read resources. 
When I accidentally found it I added tag ops to the test.

brminich commented:
AM ops are checked as well

yosefe commented on test/gtest/uct/uct_test.cc:
```diff
@@ -616,6 +616,33 @@ uct_test::entity* uct_test::create_entity(uct_iface_params_t &params) {
+                               (ucs_empty_function_return_success) :
+                               rndv_cb;
+    iface_params.rndv_arg    = rndv_arg;
+    entity *new_ent = new entity(*GetParam(), m_iface_config, &iface_params,
```
indent on column?

yosefe commented on test/gtest/uct/test_tag.cc:
```diff
@@ -1041,27 +1028,14 @@ void test_tag_mp_xrq::init()
 
-    entity *receiver = uct_test::create_entity(params);
+    entity *receiver = uct_test::create_entity(unexp_eager, unexp_rndv,
+                                             reinterpret_cast<void*>(this),
```
indentation seems off

yosefe commented:
> Initially I did not change tag operation checks in DC and they were blocked by lack of read resources.
> When I accidentally found it I added tag ops to the test.

Got it, 10x

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 21-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/737d2316d222095b82f7071aedffe695fc1c6fb4/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/b2851ce425d0ddc4c6ee88dbee56e0729bdb039a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/4a3afbd5896bd5d72722ab89750799a883c4c209/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/fb92fb272edac29187ea23f9d9fb03ca1522f723/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/654e582adbdec62ea067ca6c34f114a9e3e56bc3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/bfb2d2f9aa7bce8c09ba70bd59184cf88c6f3e27/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/ffa396a49d2d4cd1e296582186ddb62c308c3be9/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/e682ee8168c842e807368afe340796976156a427/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/59e8dd4cb0de7421347e7afc3be19da9c690344d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/767017a504ce294ac53c6ae17a9fbf413481f779/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/ddb65170259728f227f80066ac7e0f77018dc6c9/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/defb879d49fa1c386b0ccbfaf2aa2676cf52eda7/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/ec906469c9d575dd692a01fd67e12d4303d2c67e/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/8d8cde9adb49c0c423076721e9fe9a55589e5779/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/438c3c5b9ade4e006ffbb8c0c530259bb6ac544e/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/30fdbc1326cd35ed8a1fc9a486a2d65a7fc9c439/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/ec986094732d9e83b595ecac29efa6ea88ebf845/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/1fe883b28df8eb36606767756f0e4ceb908dc8d8/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/888fc81e620bd77093701426e8b255b91e81e272/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/52d1b228c0723fc18a6ea9895643ef1e3b083dda/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/c674f090f3d46b12fd1263bf16d052dde1c8830a/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/0ed580f4f86940087ae7ae1e7566ea1574b124e1/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/aebde8647394d00fd7d3cb067ea6f48cc75340b7/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/c2de7cb8fe1a3007135012f2dfc0a25cf3062cbc/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a7c4d88cc92f54031fd2c02132d0c8fc/raw/43c7543135d8f6cd59ef77989963a6b82823b202/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

yosefe commented:
/azp run
